File: split.js
const fs = require('fs');

let content = fs.readFileSync('f:/New folder (3)/WhatsAppClone/index.html', 'utf8');

// Extract styles
const styleMatch = content.match(/<style>([\s\S]*?)<\/style>/);
if (styleMatch) {
    fs.writeFileSync('f:/New folder (3)/WhatsAppClone/style.css', styleMatch[1].trim(), 'utf8');
    content = content.replace(styleMatch[0], '<link rel="stylesheet" href="style.css">');
}

// Extract main script
const scriptMatch = content.match(/<script>\s*(\/\/ --- Globals & State ---[\s\S]*?)<\/script>/);
if (scriptMatch) {
    fs.writeFileSync('f:/New folder (3)/WhatsAppClone/app.js', scriptMatch[1].trim(), 'utf8');
    content = content.replace(scriptMatch[0], '<script src="app.js"></script>');
}

fs.writeFileSync('f:/New folder (3)/WhatsAppClone/index.html', content, 'utf8');
console.log("Split completed successfully.");

File: split.py
import re

with open('f:/New folder (3)/WhatsAppClone/index.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract styles
style_match = re.search(r'<style>([\s\S]*?)</style>', content)
if style_match:
    css_content = style_match.group(1).strip()
    with open('f:/New folder (3)/WhatsAppClone/style.css', 'w', encoding='utf-8') as f:
        f.write(css_content)
    # Replace <style>...</style> with <link rel="stylesheet" href="style.css">
    content = content[:style_match.start()] + '<link rel="stylesheet" href="style.css">' + content[style_match.end():]

# Extract main script
script_match = re.search(r'<script>\s*(// --- Globals & State ---[\s\S]*?)</script>', content)
if script_match:
    js_content = script_match.group(1).strip()
    with open('f:/New folder (3)/WhatsAppClone/app.js', 'w', encoding='utf-8') as f:
        f.write(js_content)
    # Replace <script>...</script> with <script src="app.js"></script>
    content = content[:script_match.start()] + '<script src="app.js"></script>' + content[script_match.end():]

with open('f:/New folder (3)/WhatsAppClone/index.html', 'w', encoding='utf-8') as f:
    f.write(content)

print("Split completed successfully.")
